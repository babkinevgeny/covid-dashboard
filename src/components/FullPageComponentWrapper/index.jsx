import React, { Component } from 'react';
import IconButton from '@material-ui/core/IconButton';
import Fullscreen from '@material-ui/icons/Fullscreen';
import FullscreenExit from '@material-ui/icons/FullscreenExit';
import PropTypes from 'prop-types';
import './index.scss';

class FullPageComponentWrapper extends Component {
  constructor(props) {
    super(props);
    this.state = {
      fullScreen: false,
    };
  }

  onExpandButtonClick = () => {
    const { fullScreen } = this.state;
    document.body.style.overflow = fullScreen ? 'unset' : 'hidden';

    this.setState((state) => ({
      fullScreen: !state.fullScreen,
    }));
  }

  render() {
    const { children } = this.props;
    const { fullScreen } = this.state;
    const icon = fullScreen ? <FullscreenExit /> : <Fullscreen />;
    return (
      <div className={`component_wrapper${fullScreen ? ' wrapper_fullscreen' : ''}`}>
        <IconButton
          className="page_expand"
          onClick={this.onExpandButtonClick}
          id="fullScreen"
        >
          {icon}
        </IconButton>
        {children}
      </div>
    );
  }
}

FullPageComponentWrapper.propTypes = {
  children: PropTypes.element.isRequired,
};

export default FullPageComponentWrapper;
