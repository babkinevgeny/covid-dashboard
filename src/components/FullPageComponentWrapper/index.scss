.component_wrapper {
  position: relative;
}

.wrapper_fullscreen {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 2;
  background-color: white;

  & #fullScreen {
    right: 0;
    top: 0;
  }
}

