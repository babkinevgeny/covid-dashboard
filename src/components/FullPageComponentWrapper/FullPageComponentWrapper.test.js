import { shallow } from 'enzyme';
import FullPageComponentWrapper from "../FullPageComponentWrapper";

describe('FullPageComponentWrapper class component testing', () => {
  
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
    <FullPageComponentWrapper>
      <div className="div_to_check">
        <h3>Hello, World!</h3>
      </div>
    </FullPageComponentWrapper>);
  });

  it('FullPageComponentWrapper must have a correct snapshot', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('FullPageComponentWrapper renders children', () => {
    expect(wrapper.exists('.div_to_check')).toBe(true);
  });

  it('FullPageComponentWrapper change full screen state', () => {
    const nonFullScreenClassFlag = wrapper.exists('.component_wrapper');
    const fullScreenClassFlagAbsent = wrapper.exists('.wrapper_fullscreen');

    wrapper.setState({fullScreen: true});

    const fullScreenClassFlag = wrapper.exists('.wrapper_fullscreen');

    expect(nonFullScreenClassFlag).toBe(true);
    expect(fullScreenClassFlagAbsent).toBe(false);

    expect(fullScreenClassFlag).toBe(true);
  });

});