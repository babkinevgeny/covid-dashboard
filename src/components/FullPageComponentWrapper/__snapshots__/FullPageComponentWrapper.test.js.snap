// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullPageComponentWrapper class component testing FullPageComponentWrapper must have a correct snapshot 1`] = `
"<div className=\\"component_wrapper\\">
  <WithStyles(ForwardRef(IconButton)) className=\\"page_expand\\" onClick={[Function (anonymous)]} id=\\"fullScreen\\">
    <Memo([object Object]) />
  </WithStyles(ForwardRef(IconButton))>
  <div className=\\"div_to_check\\">
    <h3>
      Hello, World!
    </h3>
  </div>
</div>"
`;
